Add a model for multinomial logistic regression

pobonomo opened the pull request.

bzfwunde commented:
Here are some suggested changes:
- Rename the `output_type` parameter to `prediction_method` with the following options, that mostly match the sklearn predictor method names:
  - `predict`
  - `predict_proba`
  - `decision_function`
  - `predict_onehot` (return a 0-1 vector)
- Change the `get_error` method to always work on the probabilities.  For 0/1 return values, the error would also just be 0 or 1 which is not very informative

pobonomo commented:
> * Rename the `output_type` parameter to `prediction_method` with the following options, that mostly match the sklearn predictor method names:
>   
>   * `predict`
>   * `predict_proba`
>   * `decision_function`
>   * `predict_onehot` (return a 0-1 vector)
> * Change the `get_error` method to always work on the probabilities.  For 0/1 return values, the error would also just be 0 or 1 which is not very informative

This is more involved than I thought.

Doing it for the logistic regression is easy but it has to also work for the pipeline.

bzfwunde commented on src/gurobi_ml/sklearn/logistic_regression.py:
```diff
@@ -223,6 +235,12 @@ def _two_classes_model(self, **kwargs):
+            self.gp_model.addConstr(log_result[:, 0] == self.output[:, 1])
+            self.gp_model.addConstr(self.output[:, 0] == 1 - self.output[:, 1])
+
+        self.gp_model.write("debug.lp")
```
But I don't understand the `predict_class` parameter.

pobonomo commented:
@bzfwunde I removed this predict_class parameter as we had discussed and reworked the Janos example.

It has to do some compromises for the example but it's the best I could think off.
